src/components/navigation/SearchBar.tsx

```diff
@@ -88,7 +88,7 @@ const SearchBar = ({
   const { searchWorker } = useSearchContext();
 
   // Handle search result data from search worker message
-  const performSearch = (event: MessageEvent) => {
+  const performSearch = useCallback((event: MessageEvent) => {
```

Comment:
Probably don't need to useCallback here since this is an uncomplicated fn with no deps, it just adds memory/allocation overhead: https://kentcdodds.com/blog/usememo-and-usecallback

Reply:
https://github.com/user-attachments/assets/6d950e62-6fb5-4580-b2bd-bc5e29f3230c

Oh, I added this useCallback because I noticed it was adding a new event listener for every user input character in the searchBar, which I don't think is the expected behavior. You do make a good point though - I think there is some kind of underlying cache mechanism because I did `console.time` and somewhere, somehow React is already caching the data itself. But this makes it so there is just one event listener added after the component mounts.

Reply:
Ahh that makes sense, nice

Reply:
sounds good :D

Reply:
Hmm maybe that's an issue with `useEffect` below? That's where we add the listeners

Reply:
oh wait, good point - yeah I think that would be a better solution - just remove the dependency array, since we can avoid two separate useCallbacks with that.